lib/normalize-config.js

```diff
@@ -2,32 +2,52 @@
 
 const path = require('path');
 const v = require('@mapbox/fusspot');
+const _ = require('lodash');
+const { WEBPACK_ASSETS_BASENAME } = require('./constants');
+const fs = require('fs');
 
-function normalizeConfig(rawConfig = {}, configDir) {
-  const defaults = {
-    siteBasePath: '',
-    stylesheets: [],
-    vendorModules: [],
-    environmentVariables: [],
-    postcssPlugins: [],
-    devServerHistoryFallback: false,
-    port: 8080,
-    publicAssetsPath: 'underreact-assets',
-    browserslist: ['defaults'],
-    babelPresets: [],
-    babelPlugins: [],
-    webpackLoaders: [],
-    webpackPlugins: [],
-    webpackConfigTransform: x => x
-    // `outputDirectory` defaulted below.
-    // `publicDirectory` defaulted below.
-    // `jsEntry` defaulted below.
-    // `htmlSource` defaulted below.
-    // `polyfills` defaulted below.
-    // `production` set by the CLI.
-    // `stats` set by the CLI.
-  };
+module.exports = normalizeConfig;
+
+class Urc {
+  constructor(rawConfig = {}, defaultConfig) {
+    validateConfig(rawConfig);
+    const urc = _.merge(defaultConfig, rawConfig);
```

Comment:
`merge` might mutate, so you may want to `merge({}, a, b)`

Reply:
nice catch!